### ISSUE TITLE
[Bug]: Assumes all storms in Dataset are order by season

### ISSUE THREAD

**Issue by thomasarsouze** [umr-amap/StormR#53]
### What happened?

We reported during a meeting that behaviour with the `test_dataset.nc` test file was strange: it did not provide the same results than with the previous `test_dataset.Rd` file. 

See also the reproducible exemple below that selects the wrong storms according to the `season` filter.

### What did you expect to happen?

Checking in the `Storms` function, it turns out that in [these lines of the `retrieveStorms` function](https://github.com/umr-amap/StormR/blob/604ac111f659ddc63d134311042f589b7ea7ad58/R/getStorms.R#L683-L684), we assume that all storms are stored by season (from older to latest). 

In the `test_dataset.nc`, the storms are not stored by increasing value of `season`.

I don't think we can make that kind of assumption, so we should adapt the filter to be more generic.

### Minimal Complete Verifiable Example

```Python
> sds <- defDatabase()
=== Loading data  ===
Open database... /Library/Frameworks/R.framework/Versions/4.2/Resources/library/StormR/extdata/test_dataset.nc opened
Collecting data ...
=== DONE ===
> sts <- Storms(sds, seasons = c(2015, 2016), loi = "Vanuatu")
=== Storms processing ... ===

-> Making buffer: Done
-> Searching storms from 2015 to 2016 ...
   -> Identifying Storms: 7 potential candidates...
-> Gathering storm(s) ... 
  |=================================================================================================================| 100%

=== DONE with run time 0.9747791 sec ===

SUMMARY:
(*) LOI: Vanuatu 
(*) Buffer size: 300 km
(*) Remove Tropical Depressions (< 18 m/s in sshs): yes
(*) Number of Storms: 6 
        Name - Tropical season - SSHS - Number of observation within buffer:
        PAM - 2015 - 5 - 20 
        SOLO - 2015 - 0 - 4 
        UESI - 2020 - 1 - 5 
        ULA - 2016 - 4 - 12 
        WINSTON - 2016 - 5 - 21 
        ZENA - 2016 - 2 - 8
```

### Environment

<details>

</details>

**Comment by BaptisteDlp** [comment]
Fixed in commit abfb902d3726ca3f067dc9f28a67c18bceb2e6bf

**Comment by BaptisteDlp** [comment]
Fixed sorting the data by date in `defDatabase,` no need to change` Storms()` function
